gobot/handlers/yaml_util.go

```diff
@@ -0,0 +1,139 @@
+package handlers
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type SkillYaml struct {
+	Task_description string `yaml:"task_description"`
+	Created_by       string `yaml:"created_by"`
+	Seed_examples    []struct {
+		Question yaml.Node
+		Context  yaml.Node
+		Answer   yaml.Node
+	} `yaml:"seed_examples"`
+}
+
+type KnowledgeYaml struct {
+	Task_description string `yaml:"task_description"`
+	Created_by       string `yaml:"created_by"`
+	Domain           string `yaml:"domain"`
+	Seed_examples    []struct {
+		Question yaml.Node
+		Answer   yaml.Node
+	} `yaml:"seed_examples"`
+	Document struct {
+		Repo     string   `yaml:"repo"`
+		Commit   string   `yaml:"commit"`
+		Patterns []string `yaml:"patterns"`
+	} `yaml:"document"`
+}
+
+func (prc *PullRequestCreateHandler) generateKnowledgeYaml(requestData KnowledgePRRequest) (string, error) {
+	knowledgeYaml := KnowledgeYaml{
+		Task_description: strings.TrimSpace(requestData.Task_description),
+		Created_by:       strings.TrimSpace(requestData.Name),
+		Domain:           strings.TrimSpace(requestData.Domain),
+		Seed_examples: []struct {
+			Question yaml.Node
+			Answer   yaml.Node
+		}{},
+		Document: struct {
+			Repo     string   `yaml:"repo"`
+			Commit   string   `yaml:"commit"`
+			Patterns []string `yaml:"patterns"`
+		}{
+			Repo:     strings.TrimSpace(requestData.Repo),
+			Commit:   strings.TrimSpace(requestData.Commit),
+			Patterns: strings.Split(strings.TrimSpace(requestData.Patterns), ","),
+		},
+	}
+
+	for i, question := range requestData.Questions {
+		knowledgeYaml.Seed_examples = append(knowledgeYaml.Seed_examples, struct {
+			Question yaml.Node
+			Answer   yaml.Node
+		}{
+			yaml.Node{
+				Kind:  yaml.ScalarNode,
+				Style: yaml.FoldedStyle,
+				Value: strings.TrimSpace(question),
+			},
+			yaml.Node{
+				Kind:  yaml.ScalarNode,
+				Style: yaml.FoldedStyle,
+				Value: strings.TrimSpace(requestData.Answers[i]),
+			},
+		})
+	}
+
```

Comment:
Should we add checking to make sure that the yaml has all the required properties from the request data at this stage?

Reply:
Frontend form makes sure of that, so we don't have to get these errors from backend, that will kill overall UX.